src/components/common/MapDateSelect/index.tsx

```diff
@@ -0,0 +1,88 @@
+import DateSlider from './DateSlider';
+import strings from 'src/strings';
+import { Box, Typography, useTheme } from '@mui/material';
+import { useMemo, useState } from 'react';
+
+type MapDateSelectProps = {
+  dates: Date[];
+  onChange: (newDate: Date) => void;
+};
+
+export default function MapDateSelect({ dates, onChange }: MapDateSelectProps): JSX.Element {
+  const theme = useTheme();
+
+  const monthStrings = [
+    strings.MONTH_01,
+    strings.MONTH_02,
+    strings.MONTH_03,
+    strings.MONTH_04,
+    strings.MONTH_05,
+    strings.MONTH_06,
+    strings.MONTH_07,
+    strings.MONTH_08,
+    strings.MONTH_09,
+    strings.MONTH_10,
+    strings.MONTH_11,
+    strings.MONTH_12,
+  ];
+
+  const earliestDate = useMemo(() => {
+    return dates.reduce((prev, curr) => {
+      if (prev.getTime() < curr.getTime()) {
+        return prev;
+      }
+      return curr;
+    });
+  }, [dates]);
+
+  const latestDate = useMemo(() => {
+    return dates.reduce((prev, curr) => {
+      if (prev.getTime() > curr.getTime()) {
+        return prev;
+      }
+      return curr;
+    });
+  }, [dates]);
+
+  const latestDateIndex = useMemo(() => {
+    return dates.findIndex((d) => d === latestDate);
+  }, [dates, latestDate]);
+
+  const [selectedDate, setSelectedDate] = useState(latestDate);
+
+  const handleChange = (newDate: Date) => {
+    setSelectedDate(newDate);
+    onChange(newDate);
+  };
+
+  const getDateString = (date: Date) => {
+    return `${monthStrings[date.getUTCMonth()]} ${date.getUTCFullYear()}`;
```

Comment:
Nice, we will get localized months 👍

Reply:
Oh, okay. I see what you mean. So would you suggest changing the type of `props.dates` to a `number[]`? And passing in a list of timestamps? That would simplify several places where I have to convert between dates and timestamps (sorting). Passing in the planting site timezone as a prop would probably be the way to go here as well, I'd think, and a good idea since a timezone mismatch could land the observation in the wrong month.

Reply:
Correct, timestamps passed in as props - or the date strings "yyyy-mm-dd"  as-is and you can have the component convert to timestamps. Or keep it as-is with date objects. All theoretically correct. Throughout the product we use "yyyy-mm-dd" to represent date values, maybe timestamp semantics should be local to this component? Will let you decide, just some thoughts.

Primary suggestion - support time zone, locale changes (component should re-render on locale change), and use Intl.DateTimeFormat instead of constructing the localized string ourselves.

If you want to keep the current implementation as-is and make minimal changes, Constanza added this [util](https://github.com/terraware/terraware-web/blob/main/src/utils/useTimeZoneUtils.ts#L100) to work off date objects (primarily for monitoring charts). It's more of a hack but will still work for display purposes here. With this approach you would only need the time zone prop change.

Reply:
I like the idea of using date strings (yyyy-mm-dd), if that is what is used throughout the app. The logic of converting to timestamps can occur locally within this component, hiding those implementation details away.

Reply:
@karthikbtf I'm playing around with this a little, and I'm thinking: if I use the date string as the props, then technically I can perform the calculations internally using whatever timezone I want (e.g. UTC) and they'll still come out correctly.

For example, `new Date('2023-06-01')` creates a date in UTC by default, so it's possible to do
```typescript
new Intl.DateTimeFormat('en-US', { month: 'short', year: 'numeric', timeZone: 'UTC' }).format(new Date('2023-06-01'))
```
which returns `Jun 2023` regardless. And comparing timestamps is easy by comparing timestamps produced by `Date.parse('yyyy-mm-dd')`, which is UTC by default. 

Finally, the `onChange` callback can return yyyy-mm-dd datestrings. This way, the component transparently handles dates and doesn't need to know anything about or handle timezones at all. Thoughts on this approach?

Reply:
Yes, we can absolutely use date strings for the calculations and across the board.
My bad, the time zone suggestion wasn't for date related calculations, it was strictly for displaying a UTC date in the planting site's time zone (the Intl.DateTimeFormat's timeZone option handles this display/conversion). Under the hood all dates are timestamps, the time zone is just a presentation layer on top of that.
My assumption is the observation dates are UTC and we need to display that in the planting site's time zone. If my assumption is incorrect then we don't need time zone handling.

Reply:
Your idea sounds great for now, we can iterate once we have a better idea of the nature of observation dates. Skip time zone handling.